dds/idl/langmap_generator.cpp

```diff
@@ -219,9 +219,29 @@ struct GeneratorBase
       case AST_Expression::EV_enum:
         {
           AST_Enum* e = dynamic_cast<AST_Enum*>(the_union->disc_type());
-          if (be_global->language_mapping() == BE_GlobalData::LANGMAP_CXX11) {
-            first_label << scoped(e->name()) << "::"
-              << e->value_to_name(dv.u.enum_val)->last_component()->get_string();
+          if (be_global->language_mapping() == BE_GlobalData::LANGMAP_CXX11 ||
+              be_global->language_mapping() == BE_GlobalData::LANGMAP_FACE_CXX) {
+            std::string prefix = scoped(e->name());
+            if (be_global->language_mapping() == BE_GlobalData::LANGMAP_FACE_CXX) {
+              size_t pos = prefix.rfind("::");
+              if (pos == std::string::npos) {
+                prefix = "";
+              } else {
+                prefix = prefix.substr(0, pos) + "::";
+              }
+            } else {
+              prefix += "::";
+            }
+            first_label << prefix;
+            UTL_ScopedName* default_name = e->value_to_name(dv.u.enum_val);
```

Comment:
Would it be easier to detect that the dv is not in the valid range of the enum before calling value_to_name?